### ISSUE TITLE
Briefcase gives the wrong error message when WiX isn't found

### ISSUE THREAD

**Issue by voltagex** [beeware/briefcase#378]
![image](https://user-images.githubusercontent.com/159567/80274003-a37d0380-871a-11ea-915f-c6ffe4a47249.png)

When I didn't have WiX installed (or the WIX environment variable is unset), I would have expected the longer error message from here: https://github.com/beeware/briefcase/blob/master/src/briefcase/platforms/windows/msi.py#L39

**Comment by voltagex** [comment]
Well that's interesting:
an empty Path object gets you WindowsPath('.'), but why doesn't it evaluate to False?
```
>>> if not Path():
...     print('empty path object')
...
>>> if not os.getenv('WIX', ''):
...     print('no env set')
...
no env set
```

**Comment by voltagex** [comment]
I guess you could work around this with a call to .is_absolute() on https://github.com/beeware/briefcase/blob/master/src/briefcase/platforms/windows/msi.py#L45, I don't think there's any cases where you can have relative paths in PATH.

**Comment by freakboy3742** [comment]
This is the same error reported by #373; in that case, the problem was that the terminal session needed to be restarted.

Regardless, the root cause is the same, and appears to be exactly what you describe: the MSI backend isn't correctly differentiating between "not set" and "set incorrectly" for the WIX environment variable.

I suspect the fix here won't involve dealing with `Path` objects; it will be at the level of `getenv`. Instead of looking up `WIX`, we should be catching the `KeyError` when the environment key doesn't exist. That way, we can clearly differentiate between the two problem cases. There's already an error message present for the "not installed" case, we're just not triggering it. Presumably the "current value:" error message should also be updated, to drop the `!r` repr rendering.

**Comment by freakboy3742** [comment]
See #389 for a related bug/problem.

**Comment by freakboy3742** [comment]
Reported by @maks232 on Gitter - this appears to have been resolved along with #389.
